### PR TITLE
Get rid of GOPATH in the developers setup guide

### LINKED ISSUE #380: Get rid of GOPATH in the developers setup guide
Given we are now 100% using the local vendor folder with go modules, there is no need clone the repo inside `GOPATH` any longer. We should remove this from our setup guide.

Keep in mind that the server and webapp directories still have to be in the same directory so our `Makefile` can find them.

### PR DESCRIPTION
**Pull request by jorgeruvalcaba**
Removing GOPATH in the developer guide based on #380. Also, I changed the numeration in some guides.

Fixes #380

### CODE REVIEW

**Comment by jorgeruvalcaba** [comment]
Just complete the Mattermost contribution license agreement, hope anyone could take a look at it.

**Comment by jorgeruvalcaba** [comment]
All checks passed! :)

**Comment by jorgeruvalcaba** [comment]
@amyblais could you add a `Ready for review` tag?

File: site/content/contribute/server/developer-setup/arch.md [hunk 1]
```diff
@@ -20,52 +20,55 @@
...
+6. Logout and login to effect the changes above.
 
-8. Fork https://github.com/mattermost/mattermost-server
+7. Fork https://github.com/mattermost/mattermost-server
 
-9. Clone the Mattermost source code from your fork:
+8. Clone the Mattermost source code from your fork:
 
     ```sh
     export GITHUB_USERNAME=my_username
```
Comment (hanzei) [inline site/content/contribute/server/developer-setup/arch.md]: I don't think we need this step any longer. Would you please combine this with the next step and apply this change also to the other files?

Comment (jorgeruvalcaba) [reply site/content/contribute/server/developer-setup/arch.md]: Do you think this is ok?

File: .gitignore [hunk 2]
```diff
@@ -12,4 +12,4 @@
 .DS_Store
 
 # IDE artifacts
-.idea/
+.idea/
```
[the remark this answers is not in the document]
Comment (jorgeruvalcaba) [reply .gitignore]: Done

File: site/content/contribute/server/developer-setup/arch.md (hunk 1 above)
Comment (hanzei) [reply site/content/contribute/server/developer-setup/arch.md]: Sorry for being ambiguous. I was talking about the line https://github.com/mattermost/mattermost-developer-documentation/pull/387/files#diff-082cc8bad3408e1d0d35b9a3dc4ddd19R60. Would you please revert your changes? Sorry for the confusion.

Comment (jorgeruvalcaba) [reply site/content/contribute/server/developer-setup/arch.md]: No worries, it's reverted now. Did you mean something like this?
```ssh
export GITHUB_USERNAME=my_username // remove this line
git clone https://github.com/YOUR_GITHUB_USERNAME/mattermost-server.git // change this line
```

Comment (hanzei) [reply site/content/contribute/server/developer-setup/arch.md]: Yes

**Comment by jorgeruvalcaba** [comment]
@hanzei just update the changes based on your comments.

File: site/content/contribute/server/developer-setup/arch.md [hunk 3]
```diff
@@ -20,52 +20,54 @@
...
+8. Clone the Mattermost source code from your fork:
 
     ```sh
-    export GITHUB_USERNAME=my_username
-    mkdir -p $(go env GOPATH)/src/github.com/mattermost
-    git clone https://github.com/$GITHUB_USERNAME/mattermost-server.git $(go env GOPATH)/src/github.com/mattermost/mattermost-server
+    git clone https://github.com/YOUR_GITHUB_USERNAME/mattermost-server.git
+    cd mattermost-server
+    git config core.eol lf
+    git config core.autocrlf input
```
Comment (reflog) [inline site/content/contribute/server/developer-setup/arch.md]: I don't think we should have these here. Also, why do reset on fresh clone? 🤔

File: site/content/contribute/server/developer-setup/centos.md [hunk 4]
```diff
@@ -43,37 +43,39 @@
...
 
-9. Clone the Mattermost source code from your fork:
+8. Clone the Mattermost source code from your fork:
 
     ```sh
-    export GITHUB_USERNAME=my_username
-    mkdir -p $(go env GOPATH)/src/github.com/mattermost
-    git clone https://github.com/$GITHUB_USERNAME/mattermost-server.git $(go env GOPATH)/src/github.com/mattermost/mattermost-server
+    git clone https://github.com/YOUR_GITHUB_USERNAME/mattermost-server.git
+    cd mattermost-server
```
Comment (reflog) [inline site/content/contribute/server/developer-setup/centos.md]: same comment as above

File: site/content/contribute/server/developer-setup/arch.md (hunk 3 above)
Comment (jorgeruvalcaba) [reply site/content/contribute/server/developer-setup/arch.md]: @reflog do you mean only this line? or both `git config core.*` lines?

Comment (reflog) [reply site/content/contribute/server/developer-setup/arch.md]: lines 61-64

File: site/content/contribute/server/developer-setup/centos.md (hunk 4 above)
Comment (reflog) [reply site/content/contribute/server/developer-setup/centos.md]: everything below `clone`

File: site/content/contribute/server/developer-setup/arch.md (hunk 3 above)
Comment (jorgeruvalcaba) [reply site/content/contribute/server/developer-setup/arch.md]: done

File: site/content/contribute/server/developer-setup/centos.md (hunk 4 above)
Comment (jorgeruvalcaba) [reply site/content/contribute/server/developer-setup/centos.md]: done

**Pull request merged**